debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -500,6 +500,7 @@ public void commitOffset(Map<String, ?> sourcePartition, Map<String, ?> offset)
             for (SqlServerChangeTable table : changeTablesToBeDeleted) {
                 try {
                     dataConnection.deleteChangeTable(partition.getDatabaseName(), table);
+                    dataConnection.commit();
```

Comment:
Not sure if we need to commit the transaction here. I've added it to speed up the tests so that the deletion is visible to other db sessions faster. In the production case IMO it's not critical as long as the commit happens at some point in time.

Reply:
On the one hand, it may be important to commit the transaction right after the deletion. Otherwise, if the task fails before the commit, the transaction will get rolled back.

On the other (where the "events" semantics comes into play vs. the imperative semantics), it may be the responsibility of the stored procedure being called to commit the transaction. The connector should just notify the database that it has finished reading the table.

Let's keep it as is for now and refine it during the implementation of the real SP.